### PR TITLE
feat: simultaneously test example

### PR DESCRIPTION
**Pull request by Eomm**
Ref #193 

Details https://github.com/fastify/fast-json-stringify/issues/193#issuecomment-703804249

### CODE REVIEW

**Comment by mcollina** [comment]
CI is failing, am I missing anything?

**Comment by Eomm** [comment]
This PR want to understand the solution for the issue.

Right now you commented on the issue:

> If type: ['object', 'string'], an object with toString() will be rendered as an object. If type: ['string', 'object'], we check typeof obj === 'string' || typeof obj === 'object' && Object.prototype.hasOwnProperty.call(obj, 'toString').

I'm not sure it will work because our case is: "there is an object and I want to render it like a string sometimes and like an object sometimes". The test added show this example

File: index.js [hunk 1]
```diff
@@ -1199,7 +1199,7 @@ function nested (laterCode, name, key, location, subKey, isArray) {
 
           if (type === 'string') {
             code += `
-              ${index === 0 ? 'if' : 'else if'}(obj${accessor} === null || typeof obj${accessor} === "${type}" || obj${accessor} instanceof Date || typeof obj${accessor}.toISOString === "function" || obj${accessor} instanceof RegExp)
+              ${index === 0 ? 'if' : 'else if'}(obj${accessor} === null || typeof obj${accessor} === "${type}" || obj${accessor} instanceof Date || typeof obj${accessor}.toISOString === "function" || obj${accessor} instanceof RegExp || typeof obj${accessor} === "object")
```
Comment (mcollina) [inline index.js:1202]: I think you are missing a `hasOwnProperty` call here.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Pull request merged**